### DISCUSSION TITLE
Scheduled workflow missing github.event.repository context

### DISCUSSION THREAD

**Discussion by dhimmel** [community/community#12269] [Actions] [answered]
On GitHub actions, the `github.event` context for a scheduled job is like:

```json
{
  "event": {
    "schedule": "55 * * * *"
  }
}
```

Note that it is missing `repository`, with critical information about the repository that exists for `push`, `pull_request`, and `workflow_dispatch` jobs (but not `schedule`). Don't scheduled jobs exist within the scope of a repository just like the other event types? Would it be possible to add `github.event.repository`?

There are a couple reports in the community forum related to this issue:

- [Schedule workflow missing github.event.repository context](https://github.community/t/schedule-workflow-missing-github-event-repository-context/212293)
- [Github.event.repository.fork context does detect forks in scheduled jobs](https://github.community/t/github-event-repository-fork-context-does-detect-forks-in-scheduled-jobs/185925)

**Comment by ethomson** [comment]
Thanks for the feedback, we'll take a look! 👀

**Reply by dhimmel** [reply]
@ethomson I'm taking another look at this issue, but the pre-existing community forum links no longer seem to work. Is this content gone or archived somewhere?

```
https://github.community/t/schedule-workflow-missing-github-event-repository-context/212293
https://github.community/t/github-event-repository-fork-context-does-detect-forks-in-scheduled-jobs/185925
```

**Reply by andrewakim** [reply]
Hi @dhimmel - I'm the new PM on Actions.  I'm digging into this now and have a few quick questions.  Would you be open to jumping on a short call to chat?  Here is a link to my Calendly (edit: already setup so removed the link).

**Comment by HariSekhon** [comment]
+1

Please add this repository context info into scheduled workflows.

Without this context, you cannot selectively run workflows or steps based on fork status in `github.event.repository.fork` or other context details.

**Comment by dhimmel** [comment]
# `github.repository_owner` workaround

I created a repo to dump workflow contexts for various event types at <https://github.com/dhimmel/dump-actions-context>. Here are some of the fields beneath the `github` context for a scheduled event ([workflow logs](https://github.com/dhimmel/dump-actions-context/runs/7863389714?check_suite_focus=true#step:2:52)):

```json
  "ref": "refs/heads/main",
  "repository": "dhimmel/dump-actions-context",
  "repository_owner": "dhimmel",

  "event_name": "schedule",
  "event": {
    "schedule": "51 * * * *"
  },
```

So while the `event` context is still incomplete for `scheduled` jobs, some use cases might be able to use some of the top-level variables like `repository` and `repository_owner`. For example, add the following job condition to suppress the scheduled workflow on forks:

```yaml
if: github.repository_owner == 'dhimmel'  # replace dhimmel with the upstream user/org
```

An example of this approach in practice at https://github.com/monarch-initiative/mondo/pull/5279.

**Comment by andrewakim** [comment, marked as answer]
Hello! Just wanted to let ya'll know this feature is now live.  Check out [the changelog](https://github.blog/changelog/2022-09-27-github-actions-additional-information-available-in-github-event-payload-for-scheduled-workflow-runs/).  Thanks!

**Reply by dhimmel** [reply]
From the changelog:

> Additional information has been added to the payload of `github.event` for scheduled runs. Before this change, `github.event` for scheduled runs would only include the cron schedule. This change adds information about the repository, organization, and enterprise (when available).

🚀 and thanks @andrewakim for getting this across the finish line.

**Reply by bdzanko17** [reply]
When this will be available for Github enterprise?
